core/gf/gf.py

```diff
@@ -112,7 +112,12 @@ def parse_text(name, content, text):
     logger.debug("Grammar: {}".format(grammar))
     if grammar:
         logger.info("Parsing")
-        return [(k, [str(e) for p, e in concrete.parse(text)])
-                for k, concrete in grammar.languages.items()]
+        for k, concrete in grammar.languages.items():
+            try:
+                parse_result = concrete.parse(text.encode("UTF-8"))
+            except Exception as ex:
+                logger.exception(ex)
```

Comment:
The error message should come back with response, as previously. Now I get `{:results []}` with no indication of error. Should be something like this: `{:results [] :error "ParseError: Unexpected token: \"located\""}`

Reply:
Now I can get both `{:errors [Unexpected token: "located"], :results []}` and `{:error "\ntmp/tmpHzR1Rf/Lexicon.gf:4:5:\n   syntax error\n"}`.
Should only be :error key with string, as before.

Reply:
Can you give request example (curl) to produce both errors at same time?

Reply:
Multiple errors there are because we're iterating n x m rows, instead of crashing on first error, it returns everything that's wrong. And even with all errors, it can still produce valid result in one of the cases

Reply:
```
curl --header "Content-Type: application/json" \
--request POST \
--data '{"module":"Lang","instance":"","content":{"Lang":"\nabstract Lang =\n  Grammar,\n  Lexicon\n  ** {\n  flags startcat=Text ;\n  } ;\n","LangEng":"\nconcrete LangEng of Lang =\n  GrammarEng,\n  CustomLexiconEng\n  ** {} ;\n","Lexicon":"abstract Lexicon = Cat ** {\nfun\n  something_N : N ;\n  z._N : N ;\n\n}","CustomLexiconEng":"concrete CustomLexiconEng of Lexicon = CatEng **\n  open ParadigmsEng in {\n\nflags\n  optimize=values ;\n\nlin\n  something_N = regN \"something\" ;\n  z._N = regN \"z.\" ;\n\n} ;"},"text":"something something x y z."}' \
http://localhost:8001/parse
```
=>
```
{"error": "\ntmp/tmpXwKkTM/Lexicon.gf:4:5:\n   syntax error\n"}%
```

Reply:
```
curl --header "Content-Type: application/json" \
--request POST \
--data '{"module":"Lang","instance":"","content":{"Lang":"\nabstract Lang =\n  Grammar,\n  Lexicon\n  ** {\n  flags startcat=Text ;\n  } ;\n","LangEng":"\nconcrete LangEng of Lang =\n  GrammarEng,\n  CustomLexiconEng\n  ** {} ;\n","Lexicon":"abstract Lexicon = Cat ** {\nfun\n  be_V : V ;\n  locate_V : V ;\n  restaurant_N : N ;\n  river_N : N ;\n\n}","CustomLexiconEng":"concrete CustomLexiconEng of Lexicon = CatEng **\n  open ParadigmsEng in {\n\nflags\n  optimize=values ;\n\nlin\n  be_V = regV \"be\" ;\n  locate_V = regV \"locate\" ;\n  restaurant_N = regN \"restaurant\" ;\n  river_N = regN \"river\" ;\n\n} ;"},"text":"restaurant is a restaurant located by the river."}' \
http://localhost:8001/parse
```
=>
```
{"errors": ["Unexpected token: \"located\""], "results": []}
```

before:
=>
```
{"error": "Unexpected token: \"located\""}
```

Reply:
There should only be one way to return errors. Either keep it as i was with "error" key and string, or change it all to "errors", which will also require additional changes in API.

Reply:
I can simply make it crash on first error if that's favorable result.

Reply:
We can always improve this later, for the time being, I would suggest to keep it as it was, which will require less changes.